### PR TITLE
Remove unused dependencies

### PR DESCRIPTION
**Pull request by Fraggle**
## Description

Remove all dependencies flagged as "unused" by depcheck + manual search.

## Risk

Medium, relying on typechecking and build working. Testing in front-edge.

## Deploy Plan

Deploy `front-edge`, test, then Deploy to `front`

### CODE REVIEW

**Review by spolu — APPROVED** [review]
LGTM

**Review by aubin-tchoi — APPROVED** [review]
👍

File: front/package.json [hunk 1]
```diff
@@ -62,19 +56,15 @@
...
     "fast-diff": "^1.3.0",
     "formidable": "^3.5.1",
     "fp-ts": "^2.16.5",
     "fs-extra": "^11.1.1",
     "googleapis": "^118.0.0",
-    "googleapis-common": "^6.0.4",
     "hot-shots": "^10.0.0",
     "io-ts": "^2.2.20",
     "io-ts-reporters": "^2.0.1",
-    "io-ts-types": "^0.5.19",
```
Comment (spolu) [inline front/package.json:77]: Likely the culrpit @Fraggle?

File: front/package.json [hunk 2]
```diff
@@ -93,41 +82,25 @@
...
     "react-intersection-observer": "^9.13.1",
-    "react-katex": "^3.0.1",
     "react-markdown": "^8.0.7",
     "react-multi-select-component": "^4.3.4",
-    "react-p5": "^1.3.35",
-    "react-slick": "^0.29.0",
-    "react-syntax-highlighter": "^15.5.0",
     "react-textarea-autosize": "^8.4.0",
     "redis": "^4.6.8",
-    "rehype-katex": "^7.0.0",
```
Comment (spolu) [inline front/package.json:117]: quite suspicious that this is really unused along with all remark/rehype plugins

Comment (spolu) [reply front/package.json:117]: (some were moved to sparkle with the markdown component but there isn't a 1-1 mapping in particular the maths related ones)

Comment (Fraggle) [reply front/package.json:117]: From what I can see, it's only used [here](https://github.com/dust-tt/dust/blob/20696ad30fa52ae7ce0a4d7506fbadfce7da68f9/sparkle/src/components/markdown/Markdown.tsx#L6)

File: front/package.json (hunk 1 above)
Comment (Fraggle) [reply front/package.json:77]: io-ts-types is not imported in the code except in dust-tt/types, so I'm puzzled.
Trying to repro/config.

**Pull request merged**